# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGL2ContextFramebuffers.cpp
# Commit: ccfa767dba64
# Full Hash: ccfa767dba644dc193e0246eb8e8ff3377e8b8a5
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Client-side bindings mirror for precise CC, and merge similar codepaths. r=handyman
#   
#   * Context loss using RAII
#   * Move Program reflection Client-side
#   
# ==============================================================================

diff -r 71c122ac0ca7 -r ccfa767dba64 dom/canvas/WebGL2ContextFramebuffers.cpp
--- a/dom/canvas/WebGL2ContextFramebuffers.cpp	Wed Jan 08 22:19:14 2020 +0000
+++ b/dom/canvas/WebGL2ContextFramebuffers.cpp	Wed Jan 08 22:19:16 2020 +0000
@@ -106,7 +106,7 @@
 }
 
 bool WebGLContext::ValidateInvalidateFramebuffer(
-    GLenum target, const nsTArray<GLenum>& attachments,
+    GLenum target, const Range<const GLenum>& attachments,
     std::vector<GLenum>* const scopedVector,
     GLsizei* const out_glNumAttachments,
     const GLenum** const out_glAttachments) {
@@ -139,8 +139,8 @@
   }
   DoBindFB(fb, target);
 
-  *out_glNumAttachments = attachments.Length();
-  *out_glAttachments = attachments.Elements();
+  *out_glNumAttachments = attachments.length();
+  *out_glAttachments = attachments.begin().get();
 
   if (fb) {
     for (const auto& attachment : attachments) {
@@ -153,7 +153,7 @@
 
     if (!isDefaultFB) {
       MOZ_ASSERT(scopedVector->empty());
-      scopedVector->reserve(attachments.Length());
+      scopedVector->reserve(attachments.length());
       for (const auto& attachment : attachments) {
         switch (attachment) {
           case LOCAL_GL_COLOR:
@@ -182,8 +182,8 @@
   return true;
 }
 
-void WebGL2Context::InvalidateFramebuffer(GLenum target,
-                                          const nsTArray<GLenum>& attachments) {
+void WebGL2Context::InvalidateFramebuffer(
+    GLenum target, const Range<const GLenum>& attachments) {
   const FuncScope funcScope(*this, "invalidateFramebuffer");
 
   std::vector<GLenum> scopedVector;
@@ -210,7 +210,7 @@
 }
 
 void WebGL2Context::InvalidateSubFramebuffer(
-    GLenum target, const nsTArray<GLenum>& attachments, GLint x, GLint y,
+    GLenum target, const Range<const GLenum>& attachments, GLint x, GLint y,
     GLsizei width, GLsizei height) {
   const FuncScope funcScope(*this, "invalidateSubFramebuffer");
 